stdlib/asm/math/u64.masm

```diff
@@ -604,3 +604,89 @@ export.rotr
     not
     cswap
 end
+
+#! Counts the number of leading zeros of one unsigned 64-bit integer.
+#! The input value is assumed to be represented using 32 bit limbs, but this is not checked.
+#! Stack transition looks as follows:
+#! [n_hi, n_lo, ...] -> [clz, ...], where clz is a number of leading zeros of value n.
+#! This takes 44 cycles if n_hi = 0 (n < 2^32) and 43 cycles otherwise.
+export.clz
+    dup.0
+    eq.0
+
+    if.true             # if n_hi == 0
+        drop
+        u32clz
+        push.32
+        u32wrapping_add # clz(n_lo) + 32
+    else 
+        swap
+        drop
+        u32clz          # clz(n_hi)
+    end
+end
+
+#! Counts the number of trailing zeros of one unsigned 64-bit integer.
+#! The input value is assumed to be represented using 32 bit limbs, but this is not checked.
+#! Stack transition looks as follows:
+#! [n_hi, n_lo, ...] -> [ctz, ...], where ctz is a number of trailing zeros of value n.
+#! This takes 42 cycles if n_lo = 0 and 41 cycle otherwise.
+export.ctz
+    swap
+    dup.0
+    eq.0
+
+    if.true             # if n_lo == 0
+        drop
+        u32ctz
+        push.32
+        u32wrapping_add # ctz(n_hi) + 32
```

Comment:
Similar comment to the one above - and probably relevant to other procedures here too.